solr/core/src/java/org/apache/solr/search/CollapsingQParserPlugin.java

```diff
@@ -276,7 +281,19 @@ public int getCost() {
     }
 
     public String toString(String s) {
-      return s;
+      return "!collapse field=" + this.collapseField +
+          ", nullPolicy=" + getNullPolicyString(this.nullPolicy) +
+          ", groupHeadSelector=" + this.groupHeadSelector.toString() +
+          (hint == null ? "": ", hint=" + this.hint) +
+          ", size=" + this.size;
+    }
+
+    private String getNullPolicyString(int nullPolicy) {
```

Comment:
check if converting them enum makes it cleaner

Reply:
Converted this to enum